File: app/src/main/java/com/example/mvctutorial/designpattern/_08absttractfactory_guifactory/windows/WindowsGuiFatory.java
package com.example.mvctutorial.designpattern._08absttractfactory_guifactory.windows;

import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.Button;
import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.GuiFactory;
import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.TextArea;
import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.mac.MacButton;
import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.mac.MacTextArea;

public class WindowsGuiFatory implements GuiFactory {
    @Override
    public Button createButton() {
        return new WindowsButton();
    }

    @Override
    public TextArea createTextArea() {
        return new WindowsTextArea();
    }
}

File: app/src/main/java/com/example/mvctutorial/designpattern/_08absttractfactory_guifactory/windows/WindowsGuiFatory.kt
package com.example.mvctutorial.designpattern._08absttractfactory_guifactory.windows

import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.Button
import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.GuiFactory
import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.TextArea
import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.linux.LinuxButton
import com.example.mvctutorial.designpattern._08absttractfactory_guifactory.linux.LinuxTextArea

class WindowsGuiFatory: GuiFactory {
    override fun createButton(): Button = LinuxButton()

    override fun createTextArea(): TextArea = LinuxTextArea()
}